bamboolean/walkers/normalize.py

```diff
@@ -0,0 +1,77 @@
+from typing import TypeVar, Dict
+from bamboolean.node_visitor import NodeVisitor
+from bamboolean.ast import AST, Constraint, BinOp, UnaryOp, Bool
+from bamboolean.lexer import Token
+from bamboolean.tokens import NOT
+
+
+T1 = TypeVar('T1')
+T2 = TypeVar('T2')
+
+
+def flip_dict(d: Dict[T1, T2]) -> Dict[T2, T1]:
+    return {v: k for k, v in d.items()}
+
+
+def complementary_op(op):
+    flip_op = {
+        ('EQ', '=='): ('NE', '!='),
+        ('LT', '<'): ('GTE', '>='),
+        ('LTE', '<='): ('GT', '>'),
+        ('OR', 'OR'): ('AND', 'AND')
+    }
+    return {**flip_op, **flip_dict(flip_op)}[op]
+
+
+class NegateExpr(NodeVisitor):
```

Comment:
Using verbs for classes reads weird IMO. ExprNegator?

Reply:
Done